src/main/java/org/wildfly/security/credential/KeyPairCredential.java

```diff
@@ -63,4 +63,9 @@ public KeyPair getKeyPair() {
     public String getAlgorithm() {
         return keyPair.getPublic().getAlgorithm();
     }
+
+    public KeyPairCredential clone() {
+        return this;
```

Comment:
Most Key interfaces extend Destroyable, meaning that the key pairs have to be cloned too... somehow.